components/admin_console/team_channel_settings/channel/details/channel_details.jsx

```diff
@@ -153,14 +178,28 @@ export default class ChannelDetails extends React.Component {
             saveNeeded = true;
         } else {
             const promises = [];
-            if (!isPublic && channel.type === Constants.OPEN_CHANNEL) {
-                promises.push(actions.convertChannelToPrivate(channel.id));
+            const isOriginallyPublic = channel.type === Constants.OPEN_CHANNEL;
+
+            if (isPublic === isOriginallyPublic) {
+                promises.push(actions.patchChannel(channel.id, {
+                    ...channel,
+                    group_constrained: isSynced,
+                    type: isPublic ? Constants.OPEN_CHANNEL : Constants.PRIVATE_CHANNEL,
```

Comment:
why the type is needed here if the original one and the current one are the same?